### PR TITLE
Added ability to check duplication of rows by specific function

### PR DESCRIPTION
**Pull request by maZahaca**
Now we can pass in rule:
```js
duplicateChecker = function (_id, rows) {
   return true; // duplicate found
}
```

#51

### CODE REVIEW

**Comment by maZahaca** [comment]
@jifeon we definitely need node 4.x
```js
Parser #parse parse single page with duplicateChecker which returns true:
     TypeError: Object [object Object],[object Object],[object Object] has no method 'findIndex'
      at Parser._parseCollectionRule (lib/Parser.js:389:36)
      at Parser._parseRow (lib/Parser.js:352:14)
      at Parser.<anonymous> (lib/Parser.js:317:22)
      at Array.0 (node_modules/vow/lib/vow.js:711:39)
      at Object.callFns [as _onImmediate] (node_modules/vow/lib/vow.js:23:35)
```

**Comment by maZahaca** [comment]
@jifeon for review

**Comment by maZahaca** [comment]
The function actually should work with Promises

**Comment by maZahaca** [comment]
However, it should work like that with Promises

File: lib/Parser.js [hunk 1]
```diff
@@ -389,11 +412,33 @@ Parser.prototype = {
...
+                .then(function (parsed) {
+                    if (needDuplicateCheck && parsed.name === '_id') {
+                        return duplicateChecker(parsed.value);
+                    }
+                    return vow.resolve(false);
+                })
+                .then(function (duplicateDetected) {
+                    if (duplicateDetected) {
+                        var error = new Error('Duplicate row detected. Skip parsing.');
+                        error.code = 11;
```
Comment (jifeon) [inline lib/Parser.js]: WTF!? error.name = "DuplicationException";

**Pull request merged**